src/email/email.js

```diff
@@ -76,23 +76,41 @@ const getUserIdOrAlias = function(localPart) {
  * Send the inbound mail's contents to the corresponding rooms.
  * @param {string}  text    The text content of the email.
  * @param {ParsedMailbox}  fromAdd Email address of the sender.
- * @param {string}  alias   Destination room alias.
+ * @param {address} from    The from address object from email header.
  * @param {object}  config  Bridge configurations.
  * @returns {Promise<void>}
  */
-async function handleMail(text, fromAdd, alias, config) {
+async function handleMail(text, toAdd, fromAdd, from, config) {
+    let alias = "", receivedAddress;
+    try {
+        receivedAddress = getRoomAliasFromEmailTo(toAdd, config.bridge.domain);
+        if (receivedAddress[0] === "room") {
+            alias = `#${receivedAddress[1]}:${receivedAddress[2]}`;
+        }
+        else if (receivedAddress[0] === "user") {
+            // TODO: send message to user(DM)
+            return;
+        }
+    }
+    catch (ex) {
+        log.error(`Error resolving room address: ${ex}`);
+        //TODO: Send a reply to the sender indicating `unable to resolve room`
+        throw Error("Unable to resolve the room");
+    }
     if (!text.trim().length) {
         // text only contains whitespace (ie. spaces, tabs or line breaks)
         log.warn("Inbound email contains whitespace only");
-        return;
+        throw Error(`Inbound email contains whitespace only, ignoring this empty message from ${fromAdd.address}`);
     }
 
     log.info("Inbound email contents: "+ text.substring(0, 10) + "... ");
+    log.info("Email contents from", fromAdd.address, "will be sent to", `${receivedAddress[1]}:${receivedAddress[2]}`);
 
     const intent = bridge.getIntent(`@_email_${fromAdd.local }_${fromAdd.domain}:${config.bridge.domain}`);
+    const displayName = from.value[0].name !== "" ? `${from.value[0].name} (email)` : `${fromAdd.address} (email)`;
+    await intent.setDisplayName(displayName);
```

Comment:
I'd catch, log, but continue here. It's possible that the HS might reject your request but you shouldn't fail to handle the email because the displayname broke :)